(choco-upgrade-all-at) SchTasks improvements

MeadMA opened the pull request:
These commits:

- Add /F to SchTasks to prevent a potential execution-halting prompt
- Set the task to run elevated as SYSTEM using native Scheduled Task options, removing the need for the external 'elevate' utility
- Remove the nuspec 'elevate' dependency that is no longer required

bcurran3 commented:
MeadMA,

Thanks!

(I haven't looked at this yet - just responding due to shock that I actually go the notification.)

Bare with me as I've never done a pull or push or anything on GitHub before other than throwing a copy of my packages up and have no experience how it works, yet!

bcurran3 commented:
I'm liking what I see and appreciate the detailed commentary.

I'll see about implementing and testing it soon as I have time. (Probably not until the weekend.)

I'm still banging my head against the wall with parsing the time option as well as just understanding the right syntax of sending the time via the --parms switch. I need to fully understand the latter to make the former work. :)

If you want, I can put you on maintainership of the package and we can do it together.

MeadMA commented:
Sounds good, sir.  I was actually already looking at that.  I think you might just need to wrap $PassedParameter in $(), but I'm going to do some testing to be sure.

I just noticed I forgot to remove the "el" part for the /TR parameter.  You'll definitely need to do that when you do your testing of this pull request since the "elevate" dependency is no longer set.  If you accept the pull, be sure to add another commit to remove that "el" part.  Sorry about that oversight.

MeadMA commented:
Sorry, disregard.  I added a commit to remove the 'el' reference.  I also added the --confirm switch to the upgrade command to keep it from getting stuck waiting for user input.  It may be the weekend before I get a chance to figure out the parameter issue.

MeadMA commented:
The parameter passing issue should be fixed now.  It just needed wrapped in $().  Lesson learned in testing: parameter passing fails altogether if you run "choco install <nupkg_path>" or use "choco install --file".  See <https://github.com/chocolatey/choco/issues/1155>.  If testing locally, you need to use "choco install <package_name> --source <nupkg_path>".

I didn't update the nuspec to include instructions on the parameter passing.  This is mandatory: <https://chocolatey.org/docs/how-to-parse-package-parameters-argument#step-2---add-package-parameters-to-the-description>.  If you don't get a chance to before me, I'll go ahead and add the instructions.  It's just a busy weekend for me.

MeadMA commented:
Okay, managed to get the parameter instructions done in the nuspec.
